.github/workflows/ci.yaml

```diff
@@ -0,0 +1,22 @@
+---
+name: Continuous Integration
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  ci:
+    if: ${{ github.event_name == 'push' || github.event.pull_request.merged == true }}
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run tests
+        run: |
+          ./init.sh
+          docker compose up -d api
+          docker compose run --rm app poetry run pytest
```

Comment:
This is pretty neat if it also runs a Postgres container for you! Seems to as the db-dependent tests are passing.